pkg/providers/v1/instances_v2.go

```diff
@@ -63,6 +64,25 @@ func (c *Cloud) InstanceShutdown(ctx context.Context, node *v1.Node) (bool, erro
 	return c.InstanceShutdownByProviderID(ctx, providerID)
 }
 
+func (c *Cloud) getAdditionalLabels(zoneName string) (map[string]string, error) {
```

Comment:
Ya, I had considered that, but I preferred to keep encapsulate creating the additional labels in it's own method to keep the `InstanceMetadata` method as dumb as possible.

That being said, I could still pull out zone ID logic into it's own method and create one as you suggest, but I left the restructuring to when we have more than a single label.

That being said, I don't feel that strongly here if you do. One downside to having a single method for labels that calls out to other methods is that the list of inputs could potentially grow awkwardly over time.

Reply:
Why not 
```suggestion
func (c *Cloud) getZoneIDByZoneName(zoneName string) (string, error) {
```

This would make it more in line with the other functions. And since only one label is being returned anyway.

Reply:
Tx for the discussion. I did kind of a mix of all of this. I moved the zone logic out to the zones file so that this is just getting a value and setting the label.